lib/Queue/RedisQueue.php

```diff
@@ -34,7 +34,7 @@ public function __construct($redis) {
 	}
 
 	public function push(string $channel, $message) {
-		$this->redis->publish($channel, json_encode($message));
+		$this->redis->publish($this->redis->getDbNum() . '_' . $channel, json_encode($message));
```

Comment:
You shouldn't use the database number as a discriminator because if you're using a Redis cluster, there's only one database. Instead, you should use NEXTCLOUD_URL.